### PR TITLE
VB-309: add functionality for checking ended auctions and sending not…

### PR DESCRIPTION
**Pull request by dmitriy-uvin**
…ifications to it's participants

### CODE REVIEW

**Review by YevheniiBieliaiev — APPROVED** [review]

File: packages/backend/package.json [hunk 1]
```diff
@@ -20,7 +20,9 @@
     "prettier:check": "prettier --check \"*/**/*.{ts,tsx}\" --loglevel=warn",
     "prettier:fix": "prettier --write \"*/**/*.{ts,tsx}\" --loglevel=warn",
     "lint": "eslint . --ext .ts",
-    "lint:fix": "eslint . --ext .ts --fix"
+    "lint:fix": "eslint . --ext .ts --fix",
+    "auction-notifications:dev": "ts-node --transpile-only --files ./src/commands/index.ts auction-notifications",
+    "auction-notifications:prod": "node ./build/src/commands/index.ts auction-notifications"
```
Comment (Vitalii4as) [inline packages/backend/package.json:25]: It will not work

File: packages/backend/package.json [hunk 2]
```diff
@@ -43,6 +45,7 @@
     "socket.io": "^4.5.2",
     "swagger-jsdoc": "^6.2.5",
     "swagger-ui-express": "^4.5.0",
+    "ts-node": "^10.9.1",
```
Comment (Vitalii4as) [inline packages/backend/package.json:48]: It must be in devDependencies

**Review by Vitalii4as — CHANGES_REQUESTED** [review]
Use https://www.npmjs.com/package/node-schedule

**Comment by Vitalii4as** [comment]
Or this https://www.npmjs.com/package/cron

File: packages/backend/package.json (hunk 1 above)
Comment (d-uvin) [reply packages/backend/package.json:25]: why?

File: packages/backend/src/scheduler/index.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+export const setJob = (
+  id: string,
+  utcDate: Date | string,
+  callback: () => void,
+) => {
+  schedule.scheduleJob(id, utcDate, callback);
+};
+
+export const initAuctionJobs = (productRepository: ProductRepository) => {
```
Comment (Nightlngale) [inline packages/backend/src/scheduler/index.ts:21]: How would you set a job for created in runtime auction. I believe,  it make sense to have a service with Api to add a job.

**Review by Nightlngale — CHANGES_REQUESTED** [review]

File: packages/backend/package.json [hunk 4]
```diff
@@ -26,6 +26,7 @@
     "@aws-sdk/client-s3": "^3.145.0",
     "@prisma/client": "^4.1.1",
     "@types/jsonwebtoken": "^8.5.8",
+    "@types/node-schedule": "^2.1.0",
```
Comment (Nightlngale) [inline packages/backend/package.json:29]: dev deps

File: packages/backend/src/services/product.ts [hunk 5]
```diff
@@ -30,6 +30,7 @@ import { FieldError } from 'error/product/field-error';
 import { createPostSchema, updatePostSchema } from 'validation/product/schemas';
 import { NotVerifiedError } from 'error/user/not-verified';
 import { lang } from '@lang';
+import { createAuctionJob, updateAuctionJob } from '@scheduler';
```
Comment (Nightlngale) [inline packages/backend/src/services/product.ts:33]: inject it via constructor

**Review by Nightlngale — CHANGES_REQUESTED** [review]

**Review by Vitalii4as — APPROVED** [review]

**Pull request merged**